### PR TITLE
Fix `_log_complete_trial` for constrained optimization

### PR DESCRIPTION
**Pull request by not522**
## Motivation

`_log_complete_trial` assumes that `study.best_trial` does not raise `ValueError` because the latest trial is `COMPLETE` (Note: if no trial is `COMPLETE`, `best_trial` raises `ValueError`). But, after #5426 was merged, it raises `ValueError` if no trial is **feasible**.

## Description of the changes

This PR added try-except for the constrained optimization and fixed the logging message.

### CODE REVIEW

**Comment by not522** [comment]
@c-bata @HideakiImamura Could you review this PR?

File: optuna/study/study.py [hunk 1]
```diff
@@ -1111,27 +1111,30 @@ def _log_completed_trial(self, trial: FrozenTrial) -> None:
...
-                    best_trial.number,
-                    best_trial.value,
-                )
+
+            message = (
+                f"Trial {trial.number} finished with value: {trial_value} and parameters: "
+                f"{trial.params}."
             )
+            if best_trial is not None:
+                message += f" Best is trial {best_trial.number} with value: {best_trial.value}."
```
Comment (c-bata) [inline optuna/study/study.py:1136]: A nit: I would say we can simplify the code a little as follows:

```suggestion
            try:
                best_trial = self.best_trial
                message += f" Best is trial {best_trial.number} with value: {best_trial.value}."
            except ValueError:
                # If no feasible trials are completed yet, study.best_trial raises ValueError.
                pass
```

**Review by c-bata — APPROVED** [review]
LGTM with a nit.

**Comment by c-bata** [comment]
Should we add a bug label, by the way? The problem was introduced in #5426, which is not released yet.

**Comment by not522** [comment]
Thank you for your review. I updated it as you suggested.

> Should we add a bug label, by the way? The problem was introduced in #5426, which is not released yet.

Not a strong opinion, but as a type of PR it is a bug fix, so I think it can stay as is.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**